client/app/pages/dashboards/public-dashboard-page.js

```diff
@@ -52,6 +53,8 @@ const PublicDashboardPage = {
         };
 
         this.extractGlobalParameters();
+      }).catch((error) => {
+        throw new PromiseRejectionError(error);
```

Comment:
Do we have deliberate relevant exceptions where we could insert meaningful error messages to?

Reply:
`PromiseRejectionError` is actually such an exception: https://github.com/getredash/redash/blob/master/client/app/lib/promise-rejection-error.js

Reply:
I mean the `error` we catch - make it's message be meaningful.
"...page cannot be found" is just confusing in this case.

Reply:
What would you expect here?

Reply:
Oh I see there's a relevant error message a few lines above:
https://github.com/getredash/redash/blob/2ca763747a787d8a9af90189953440745fcfa238/client/app/pages/dashboards/public-dashboard-page.js#L45

 BUT it seems to be doing it wrong, as PromiseRejectionError requires
1. `rejection.status !== undefined/404/403`
2. `rejection.data.error = 'the message'`

Reply:
But this is not what this change about. This change about showing a 404 when you use a bad token (either non existing or just invalid).

Reply:
Ok 😭